core/src/main/java/org/mule/context/DefaultMuleContextBuilder.java

```diff
@@ -92,6 +95,25 @@
 
     protected SplashScreen shutdownScreen;
 
+    public DefaultMuleContextBuilder()
+    {
+        this(false);
+    }
+
+    /**
+     * if {@code createSimpleRegistry} is {@code true}, then a {@link SimpleRegistry}
+     * will automatically be added to the created context. That only makes sense
+     * when trying to start a light weight context which doesn't require starting a heavier
+     * registry such as Spring or Guice or when testing. This is not something we would
+     * advice on production
+     *
+     * @param createSimpleRegistry whether or not to add a {@link SimpleRegistry} to the created context
+     */
+    public DefaultMuleContextBuilder(boolean createSimpleRegistry)
```

Comment:
I agree.. I have it on my todo list to remove per comments from @dfeist too. However, that's easier said than done because of the MMC agent. Will work on that